pose_evaluation/metrics/embedding_distance_metric.py

```diff
@@ -0,0 +1,224 @@
+from typing import Literal, List, Union
+import logging
+
+import torch
+from torch import Tensor
+from torch.types import Number
+import numpy as np
+from sentence_transformers import util as st_util
+
+from pose_evaluation.metrics.base_embedding_metric import EmbeddingMetric
+
+
+# Useful reference: https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/util.py#L31
+# * Helper functions such as batch_to_device, _convert_to_tensor, _convert_to_batch, _convert_to_batch_tensor
+# * a whole semantic search function, with chunking and top_k
+
+# See also pgvector's C implementation: https://github.com/pgvector/pgvector/blob/master/src/vector.c
+# * cosine_distance: https://github.com/pgvector/pgvector/blob/master/src/vector.c#L658
+# * l2_distance https://github.com/pgvector/pgvector/blob/master/src/vector.c#L566
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+ValidDistanceKinds = Literal["cosine", "euclidean", "manhattan", "dot"]
+TensorConvertableType = Union[List, np.ndarray, Tensor]
+
+
+class EmbeddingDistanceMetric(EmbeddingMetric):
+    def __init__(
+        self,
+        kind: ValidDistanceKinds = "cosine",
+        device: Union[torch.device, str] = None,
+        dtype=torch.float32,
```

Comment:
shouldn't this default to None? (later you check None)

Reply:
So it should, thank you